File: src/temp/primitives/types/PCharacter.kt
package temp.primitives.types

import temp.primitives.Primitive
import temp.primitives.PrimitiveType

class PCharacter(var value: Char) : Primitive() {
    override fun getPrimitiveType(): PrimitiveType {
        return PrimitiveType.PR_CHARACTER
    }

    override fun isPrimitiveType(primitiveType: PrimitiveType): Boolean {
        return primitiveType == PrimitiveType.PR_CHARACTER
    }

    override fun getPrimitiveValue(): Any {
        return this.value
    }

    override fun negate(): Primitive {
        return PNone()
    }

    override fun not(): Primitive {
        return PNone()
    }

    override fun plus(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun minus(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun times(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun div(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun rem(primitive: Primitive): Primitive {
        return PNone()
    }

    // TODO Chars Cannot Be Compared For Now
    override fun compareGreater(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun compareLesser(primitive: Primitive): Primitive {
        return PNone()
    }

    override fun toString(): String {
        return value.toString()
    }
}

File: src/codepulsar/primitive/type/PCharacter.cpp
#include "PCharacter.h"


Pulsar::PCharacter::PCharacter(char value) {
    this->value = value;
}

Pulsar::PrimitiveType Pulsar::PCharacter::getPrimitiveType() {
    return PR_CHARACTER;
}

bool Pulsar::PCharacter::isPrimitiveType(PrimitiveType primitiveType) {
    return primitiveType == PR_CHARACTER;
}

std::any Pulsar::PCharacter::getPrimitiveValue() {
    return this->value;
}

Pulsar::Primitive* Pulsar::PCharacter::unaryNegate() {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::unaryNot() {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::plus(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::minus(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::times(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::div(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::rem(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::compareGreater(Primitive* primitive) {
    return new PNone();
}

Pulsar::Primitive* Pulsar::PCharacter::compareLesser(Primitive* primitive) {
    return new PNone();
}

std::string Pulsar::PCharacter::toString() {
    return std::to_string(this->value);
}
